lib/Elastica/Transport/Guzzle.php

```diff
@@ -115,12 +115,14 @@ public function exec(Request $request, array $params)
      */
     protected function _createPsr7Request(Request $request, Connection $connection)
     {
+        $headers = $connection->hasConfig('headers') && \is_array($connection->getConfig('headers'))
```

Comment:
What is the `\` doing here before is_array?

Reply:
Just a micro-optimisation added by my linter, I'll remove it when I'm adding a test if you'd prefer.

Reply:
Can you share on how this optimises the code? Want to learn more here.

Reply:
I first learned of it when I started using the [EA Inspections plugin](https://github.com/kalessil/phpinspectionsea) for PHPStorm as one of the things it suggests is to fully qualify global function calls.

There's a writeup and some benchmarks [here](https://veewee.github.io/blog/optimizing-php-performance-by-fq-function-calls/), and a few packages for automatically dealing with it - one from [Roave / ocramius](https://github.com/Roave/FunctionFQNReplacer) and one from [nilportugues](https://github.com/nilportugues/php-backslasher).

Granted it's a tiny tiny tiny optimisation but it's become part of my workflow.

Reply:
Interesting, was not aware of this optimisation. Thanks for the details, I would probably skip it in this PR and rather have a follow up PR that adds it in all parts of the code base to be consistent. It seems that is something that could be done with https://github.com/nilportugues/php-backslasher in an automated way?